### PR TITLE
LPS-184641

### PR DESCRIPTION
**Pull request by ces-quanhuynh**
Hi team,
Please help me review this PR.

# Motivation
In this PR, I solve 1 tasks:
[LPS-184641](https://issues.liferay.com/browse/LPS-184641) [BE] Upgrade processs to update the existing folders
# Proposed Solution
I overwrite the doUpgrade to add the new column "allowedPropaganda" with "tinyint" data type. After that, I execute the query to update false value for all the rows.

### CODE REVIEW

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/registry/DLWebUpgradeStepRegistrator.java [hunk 1]
```diff
@@ -49,6 +50,8 @@ public void register(Registry registry) {
 				DLPortletKeys.DOCUMENT_LIBRARY_ADMIN));
 
 		registry.register("1.0.1", "1.0.2", new UpgradePortletPreferences());
+
+		registry.register("1.0.2", "1.1.0", new DLUpgradeProcess());
```
Comment (locpham97) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/registry/DLWebUpgradeStepRegistrator.java:54]: The upgrade process focus on the DLFolder, I think it is better to change the DLFolderUpgradeProcess. Moreover,this file should in documet-library-service, not in document-library-web

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/v1_1_0/DLUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ */
+public class DLUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (!hasColumn("DLFolder", "allowedPropagation")) {
+			alterTableAddColumn("DLFolder", "allowedPropagation", "TINYINT");
+		}
+
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
```
Comment (locpham97) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/v1_1_0/DLUpgradeProcess.java:34]: using runSQL

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/v1_1_0/DLUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * @author Quan Huynh
+ */
+public class DLUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (!hasColumn("DLFolder", "allowedPropagation")) {
+			alterTableAddColumn("DLFolder", "allowedPropagation", "TINYINT");
```
Comment (locpham97) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/v1_1_0/DLUpgradeProcess.java:31]: column type is boolean. see https://github.com/liferay/liferay-portal/blob/master/modules/apps/fragment/fragment-service/src/main/java/com/liferay/fragment/internal/upgrade/v2_6_0/FragmentEntryUpgradeProcess.java#L75

Comment (ces-quanhuynh) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/v1_1_0/DLUpgradeProcess.java]: I've updated it.

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/v1_1_0/DLUpgradeProcess.java (hunk 2 above)
Comment (ces-quanhuynh) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/v1_1_0/DLUpgradeProcess.java]: I've updated it.

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/registry/DLWebUpgradeStepRegistrator.java (hunk 1 above)
Comment (ces-quanhuynh) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/registry/DLWebUpgradeStepRegistrator.java]: I've updated it.

**Comment by ces-quanhuynh** [comment]
@locpham97 
I've updated the code following your comment.
Please review again.
Thanks!

**Comment by ces-locpham** [comment]
It is good for me

**Comment by ces-locpham** [comment]
@tototrinh,Could you check in this PR?

**Review by ces-locpham — APPROVED** [review]

**Comment by tototrinh** [comment]
@locpham97 LGTM. Thanks.

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/upgrade/v3_2_8/DLFolderColumnUpgradeProcess.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (!hasColumn("DLFolder", "allowedPropagation")) {
+			alterTableAddColumn("DLFolder", "allowedPropagation", "BOOLEAN");
+		}
+
+		runSQL(
+			StringBundler.concat(
+				"update DLFolder set allowedPropagation = [$FALSE$]"));
```
Comment (tototrinh) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/upgrade/v3_2_8/DLFolderColumnUpgradeProcess.java:33]: Why do you use `StringBundler.concat` here when there is only 1 element?

**Review by ces-totrinh — COMMENTED** [review]
@ces-quanhuynh @locpham97,
Please modify the code based on my comments and update the PR. Thanks.

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/upgrade/v3_2_8/DLFolderColumnUpgradeProcess.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+package com.liferay.document.library.internal.upgrade.v3_2_8;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+/**
+ * @author Quan Huynh
+ */
+public class DLFolderColumnUpgradeProcess extends UpgradeProcess {
```
[the remark this answers is not in the document]
Comment (ces-quanhuynh) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/upgrade/v3_2_8/DLFolderColumnUpgradeProcess.java]: I've updated the code!

**Comment by ces-quanhuynh** [comment]
@tototrinh 
I've updated the code following your comment.
Please review again!
Thanks.
cc @locpham97

**Comment by ces-totrinh** [comment]
Resend PR here: https://github.com/tototrinh/liferay-portal/pull/52

**Pull request closed without merging**